### ISSUE TITLE
[enhancement] kubectl alpha debug add SYS_PTRACE for debug container

### ISSUE THREAD

**Issue by yili1992** [kubernetes/kubernetes#97103]
```
VERSION: v1.18.3
Command:  kubectl alpha debug -it xxx-854d568b99-klgc9 --image=myimage:latest --container=xxx-854d568b99-klgc9 --target=xxx
In Debug Container: I use PTRACE for my process , but I get error message: 
Failed to setns 4: Operation not permitted
```
**What would you like to be added**:
I hope that this debug feature can add "SYS_PTRACE" for container

**Why is this needed**:
We always to analysis  our process in POD , this PTRACE is very useful

**Comment by wawa0210** [comment]
It seems reasonable, we can try to declare a flag to identify this parameter. If the community accepts this proposal, I can try to contribute

**Comment by verb** [comment]
I agree, this is super useful. For ephemeral containers this will be blocked on #53188, which I hope to address in 1.21.

@wawa0210 if you're interested, you could implement this for adding a debug container with `--copy-to` and then extend it to also work for ephemeral containers when #53188 is resolved. What do you think?

For the implementation, my impulse would be to allow an arbitrary list of capabilities specified by command line, for example `kubectl debug ... --capabilities=NET_ADMIN,SYS_PTRACE`. It surfaces a bit more complexity but I think the use cases are too varied to be able to effectively address with booleans. This would be a good topic for a quick discussion at a sig-cli meeting, though.

**Comment by wawa0210** [comment]
@verb 

Thanks for the reply, I am very interested. I might get to know it first, and then start to work. If there are new ideas, I am looking forward to knowing~

/assign

**Comment by fjammes** [comment]
Is there some news about this interesting feature?

**Comment by verb** [comment]
@fjammes We updated the ephemeral containers API in 1.22 to make this possible with ephemeral containers and plan on adding support to `kubectl debug` in 1.23.

**Comment by coderplay** [comment]
@verb 

> plan on adding support to kubectl debug in 1.23.
Was it added?

**Comment by P1ng-W1n** [comment]
From what I can tell it wasn't added.
@verb Is there another ticket that could covers this activity?

**Comment by verb** [comment]
Using `kubectl debug` with `--profile=general` will add `SYS_PTRACE` permission. See https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/1441-kubectl-debug#debugging-profiles
